tests.py

```diff
@@ -231,14 +234,55 @@ def test_tld_function(self):
         psl = publicsuffix.PublicSuffixList()
         # checks that the eTLD or TLD is produced
         assert psl.get_tld('com') == 'com'
+        self.assertEqual(psl.get_tld('city.kobe.jp'), 'kobe.jp')
+        self.assertEqual(psl.get_tld('kobe.jp'), 'kobe.jp')
+        self.assertEqual(psl.get_tld('amazonaws.com'), 'amazonaws.com')
         assert psl.get_tld('telinet.com.pg', wildcard=True) == 'com.pg'
         assert psl.get_tld('telinet.com.pg', wildcard=False) == 'pg'
+        self.assertEqual(psl.get_tld('com.pg', wildcard=True), 'com.pg')
+        self.assertEqual(psl.get_tld('com.pg', wildcard=False), 'pg')
         assert psl.get_tld('telinet.co.uk', wildcard=False) == 'co.uk'
+        self.assertEqual(psl.get_tld('co.uk', wildcard=True), 'co.uk')
+        self.assertEqual(psl.get_tld('co.uk', wildcard=False), 'co.uk')
         assert psl.get_tld('blah.local', strict=True) is None
+        self.assertEqual(psl.get_tld('blah.local', wildcard=False), None)
+        self.assertEqual(psl.get_tld('blah.local'), 'local')
+
+        # the tld for empty string
+        self.assertEqual(psl.get_tld(''), None)
+        self.assertEqual(psl.get_tld('.'), '')  # XXX: Is it correct?
```

Comment:
ideally this is '.' -- in our internal use cases, I handle this as a one-off so that we don't map root requests to nothing.  since it's more of a normalization, i hadn't requested to change it in the library. we could. this happens because of the rstrip at the top. thoughts?

Reply:
In that case, it is better to return `''` than `'.'` . This is because `get_tld` and `get_sld` always return a domain string without trailing `'.'` . If you need a trailing dots, you can add it to the return value of `get_sld` .

I arrange tests to emphasize that (db5b8db5033ab46e568f8e8e78ab5bc3ef6a5240).